### ISSUE TITLE
StrictMock returning void? 

### ISSUE THREAD

**Issue by kikaragyozov** [markjsc/AutoMockHelper#5]
As per your documentation:

> StrictMock<TAnyInterfaceOrClass>() - Creates, registers and returns a mock instance of the provided interface or class using Strict behavior (causes the mock to always throw an exception for invocations that don't have a corresponding setup)

It should return an instance, much like `MockFor<>` does, allowing you to chain method calls like `.Setup` and others.

With `StrictMock`, you can't chain anything. Is this intentional, and if so, could you supply an example of its use? Thank you!

**Comment by markjsc** [comment]
Hi @SpiritBob - thanks for taking the time to post your question.

It's been quite a while since I've updated this, so I'll take some time to dig into it and get back with you. 

I suspect you're right - we could treat StrictMock just like some of the other behaviors that return an instance for chaining. If that works as expected then I'll make the update to the code and NuGet package.

Stay tuned (it may take me several days...)

**Comment by kikaragyozov** [comment]
Thank you @markjsc ! You've done a very good thing with this. It certainly makes unit testing as clean as possible, although I found something perhaps lacking in that regard.

Imagine having 10 dependencies which we mock successfully.
Now imagine that I want all the setups for the given dependencies to be validated that they have ran only once.
Just to make that work, I'll have to remove the magical call to `VerifyAll()`, and add 10 x NumberOfSetupsPerDependency huge, boiler-plate-like chained method calls, in the form of `MockFor<TDependency>().Verify(<ExpressionToBeVerifiedItRanOnlyOnce>, Times.Once)`.

The code you have to write and maintain just skyrockets. Could you perhaps add an overload to `VerifyAll` method, with a `Times` argument? Obviously, if you wanted certain setups/expressions to be validated that they ran twice, and others 3 times, you'd have to go down the long path, but that's because there's no clean way for such cases to be handled.

If you could implement this as a feature, I'll love you for it, as everything will get even cleaner than we already get it as!

**EDIT:** Perhaps it can be a separate method, rather than an overload to `VerifyAll` (`VerifyAllOnce` for example, if you feel like you shouldn't be handling other cases.)

**Comment by markjsc** [comment]
That's a great suggestion!

Naturally, any call that is supported by the underlying AutoMock library can be added to this.

I'll include your suggestions when I make the other update. (Post back if you think of any additional needs in the meantime.)

**Comment by markjsc** [comment]
Hi @SpiritBob  - I've implemented your suggestion to return the Mock<> from the StrictMock() method. Great suggestion - and the implementation was no problem.

However, your request to add different behaviors to VerifyAll() is a bit more complicated. My library just calls existing behaviors exposed by the underlying AutoMock library. In the case of VerifyAll(), an overload that includes the Times parameter would require updating the AutoMock library.

As you can see from the AutoMock source, the VerifyAll method uses a private variable (typeMap):
```
        /// <summary>
        /// This is a shortcut for calling `mock.VerifyAll()` on every mock that we have.
        /// </summary>
        public void VerifyAll()
        {
            foreach (var pair in typeMap)
            {
                if (pair.Value is MockInstance instance)
                    instance.Mock.VerifyAll();
            }
        }
```
The typeMap variable is not exposed in any way that I could reach it with my library.

Perhaps you could make this request on the AutoMock project. Once the author has added support there, then I could add it here as well.

I've updated the version to [1.1.2.0](https://github.com/markjsc/AutoMockHelper/releases/tag/1.1.2.0), including updating the [NuGet package](https://www.nuget.org/packages/AutoMockHelper/1.1.2).

Hope this helps!

**Comment by markjsc** [comment]
Marking as closed - Version [1.1.2.0](https://github.com/markjsc/AutoMockHelper/releases/tag/1.1.2.0) released.
